addon/components/mdc-tab-bar-scroller.js

```diff
@@ -0,0 +1,141 @@
+import layout from '../templates/components/mdc-tab-bar-scroller';
+import getElementProperty from '../utils/get-element-property';
+import getComponentProperty from '../utils/get-component-property';
+import styleComputed from '../utils/style-computed';
+
+const { get, computed, set, run } = Ember;
+const { cssClasses, strings } = MDCTabBarScrollerFoundation;
+
+export default Ember.Component.extend(MDCComponent, {
+  //region Ember Hooks
+  classNames: ['mdc-tab-bar-scroller'],
+  layout,
+  init() {
+    this._super(...arguments);
+    set(this, 'backIndicatorClasses', Ember.A([]));
+    set(this, 'forwardIndicatorClasses', Ember.A([]));
+    set(this, 'mdcScrollFrameStyles', {});
+  },
+  didInsertElement() {
+    this._super(...arguments);
+    set(this, 'scrollFrameElement', getElementProperty(this, 'querySelector')(strings.FRAME_SELECTOR));
+    set(this, 'tabBarElement', getElementProperty(this, 'querySelector')(strings.TABS_SELECTOR));
+    set(this, 'forwardIndicatorElement', getElementProperty(this, 'querySelector')(strings.INDICATOR_FORWARD_SELECTOR));
+    set(this, 'backIndicatorElement', getElementProperty(this, 'querySelector')(strings.INDICATOR_BACK_SELECTOR));
+  },
+  //endregion
+
+  //region Properties
+  /**
+   * @type {Object}
+   */
+  CLASS_NAMES: cssClasses,
+
+  /**
+   * @type {Object}
+   */
+  scrollFrameElement: null,
+
+  /**
+   * @type {Object}
+   */
+  tabBarElement: null,
+
+  /**
+   * @type {Object}
+   */
+  forwardIndicatorElement: null,
+
+  /**
+   * @type {Object}
+   */
+  backIndicatorElement: null,
+
+  /**
+   * @type {String[]}
+   */
+  backIndicatorClasses: null,
+
+  /**
+   * @type {String[]}
+   */
+  forwardIndicatorClasses: null,
+
+  /**
+   * @type {Ember.Component}
+   */
+  'tab-bar': null,
+
+  /**
+   * Key value pairs for CSS styles
+   * @type {Object}
+   */
+  mdcScrollFrameStyles: null,
+  //endregion
+
+  //region Methods
+  createFoundation() {
+    return new MDCTabBarScrollerFoundation({
+      addClass: (className) => run(() => get(this, 'mdcClasses').addObject(className)),
+      removeClass: (className) => run(() => get(this, 'mdcClasses').removeObject(className)),
+      eventTargetHasClass: (target, className) => target.classList.contains(className),
+      addClassToForwardIndicator: (className) => run(() => get(this, 'forwardIndicatorClasses').addObject(className)),
+      removeClassFromForwardIndicator: (className) => run(() => get(this, 'forwardIndicatorClasses').removeObject(className)),
+      addClassToBackIndicator: (className) => run(() => get(this, 'backIndicatorClasses').addObject(className)),
+      removeClassFromBackIndicator: (className) => run(() => get(this, 'backIndicatorClasses').removeObject(className)),
+      isRTL: () => getElementProperty(this, 'direction') === 'rtl',
+      registerBackIndicatorClickHandler: (handler) => get(this, 'backIndicatorElement').addEventListener('click', handler),
```

Comment:
I'm not sure if this will happen or not, but I have a concern:

If the back indicator element is un-rendered and re-rendered as a different DOM element by Ember after the click handler was registered, then it will no longer have a `click` event listener attached. This is bad, and should be avoided.

If this _is_ a problem, I see two ways to avoid it:

0. Store the back indicator element click handlers in the component JS, and re-attach them every time on `didRender`
0. Store the back indicator element click handlers in the component JS, and use the `click` Ember hook for the whole component, firing off each back indicator element click handler if the event's `target` was the back indicator element.

I think I'd prefer the latter of those two, and I'm not sure that list is exhaustive.

Reply:
I'm not sure I understand-- why would the back indicator element ever be un-rendered without the entire component itself being un-rendered?

Reply:
You're completely right, it wouldn't be. This was only a risk if there was a boolean around the indicators or something, which there's clearly not.